02_classify_compressed/run_mutation_compressed.py

```diff
@@ -0,0 +1,228 @@
+            mutation_log_df = None
+            gene = gene_series.gene
+            classification = gene_series.classification
+            progress.set_description('gene: {}'.format(gene))
+
+            try:
+                gene_dir = fu.make_output_dir(experiment_dir, gene)
+                check_file = fu.check_output_file(gene_dir,
+                                                  gene,
+                                                  shuffle_labels,
+                                                  model_options)
+                tcga_data.process_data_for_gene(gene,
+                                                classification,
+                                                gene_dir,
+                                                shuffle_labels=shuffle_labels)
+            except ResultsFileExistsError:
+                # this happens if cross-validation for this gene has already been
+                # run (i.e. the results file already exists)
+                if io_args.verbose:
+                    print('Skipping because results file exists already: gene {}'.format(
+                        gene), file=sys.stderr)
+                mutation_log_df = fu.generate_log_df(
+                    log_columns,
+                    [gene, model_options.training_data, shuffle_labels, 'file_exists']
+                )
+                fu.write_log_file(mutation_log_df, io_args.log_file)
+                continue
+            except KeyError:
+                # this might happen if the given gene isn't in the mutation data
+                # (or has a different alias, TODO we could check for this later)
+                print('Gene {} not found in mutation data, skipping'.format(gene),
+                      file=sys.stderr)
+                mutation_log_df = fu.generate_log_df(
+                    log_columns,
+                    [gene, model_options.training_data, shuffle_labels, 'gene_not_found']
+                )
+                fu.write_log_file(mutation_log_df, io_args.log_file)
+                continue
+
+            try:
+                # columns should be standardized before compression
+                # so we don't want to standardize them here
+                results = run_cv_stratified(tcga_data,
+                                            'gene',
+                                            gene,
+                                            model_options.training_data,
+                                            sample_info_df,
+                                            model_options.num_folds,
+                                            shuffle_labels,
+                                            standardize_columns=False)
+            except NoTrainSamplesError:
+                if io_args.verbose:
+                    print('Skipping due to no train samples: gene {}'.format(
+                        gene), file=sys.stderr)
+                mutation_log_df = fu.generate_log_df(
```

Comment:
This is cool. I'll need to keep this in mind. I know Milton mentioned this as something to consider for future projects